src/lib.rs

```diff
@@ -103,9 +103,22 @@ pub enum DecodeFailed {
     /// An invalid block type was found.
     InvalidBlock(u8),
 
+    /// An invalid block size was found.
+    InvalidBlockSize(u32),
+
     /// An invalid pretree element was found.
     InvalidPretreeElement(u16),
 
+    /// Invalid pretree run-length encoding.
+    // TODO: Better name?
```

Comment:
Personally I don't know if having so many specific options will be of any use. Most of the time it'll be like "okay the file's corrupted doesn't matter where". But on the other hand it doesn't hurt to have so eh.

Reply:
Actually - it wouldn't be a bad idea to throw all these errors under some catch-all `CorruptBlock(CorruptBlockError)`

Reply:
Given we're a library and not an application it's probably not worth the effort.